webhook_server.py

```
@@ -0,0 +1,131 @@
+            ["git", "pull", "origin", "main"],
+            capture_output=True, text=True, timeout=60
+        )
+        logger.info(result.stdout)
+        if result.returncode != 0:
+            logger.error(f"git pull failed: {result.stderr}")
+            return False
+        
+        # 3. docker build
+        logger.info("Step 2: docker build")
+        result = subprocess.run(
+            ["docker", "build", "-t", "new-api:latest", "."],
```

Ruff (0.15.17)

[error] Starting a process with a partial executable path

(S607)